docs/user-guide/nemosdk.rst

```diff
@@ -0,0 +1,121 @@
+
+-----------------------------------------
+Usage
+-----------------------------------------
+
+We recommend getting slightly familiar with NeMo SDK before jumping into this. The documentation can be found here.
+
+Let's walk through the example usage for how you can launch a slurm job using `examples/launch_slurm.py <https://github.com/NVIDIA/NeMo-Curator/blob/main/examples/nemo_sdk/launch_slurm.py>`_.
+
+.. code-block:: python
+
+
+    import nemo_sdk as sdk
+    from nemo_sdk.core.execution import SlurmExecutor
+
+    from nemo_curator.nemo_sdk import SlurmJobConfig
+
+    @sdk.factory
+    def nemo_curator_slurm_executor() -> SlurmExecutor:
+        """
+        Configure the following function with the details of your SLURM cluster
+        """
+        return SlurmExecutor(
+            job_name_prefix="nemo-curator",
+            account="my-account",
+            nodes=2,
+            exclusive=True,
+            time="04:00:00",
+            container_image="nvcr.io/nvidia/nemo:dev",
+            container_mounts=["/path/on/machine:/path/in/container"],
+        )
+
+First, we need to define a factory that can produce a ``SlurmExecutor``.
+This exectuor is where you define all your cluster parameters. Note: NeMo SDK only supports running on SLURM clusters with `Pyxis <https://github.com/NVIDIA/pyxis>`_ right now.
+After this, there is the main function
+
+.. code-block:: python
+
+    # Path to NeMo-Curator/examples/slurm/container_entrypoint.sh on the SLURM cluster
+    container_entrypoint = "/cluster/path/slurm/container_entrypoint.sh"
+    # The NeMo Curator command to run
+    curator_command = "text_cleaning --input-data-dir=/path/to/data --output-clean-dir=/path/to/output"
+    curator_job = SlurmJobConfig(
+        job_dir="/home/user/jobs",
+        container_entrypoint=container_entrypoint,
+        script_command=curator_command,
+    )
+
+First, we need to specify the path to `examples/slurm/container-entrypoint.sh <https://github.com/NVIDIA/NeMo-Curator/blob/main/examples/slurm/container-entrypoint.sh>`_ on the cluster.
+This shell script is responsible for setting up the Dask cluster on Slurm and will be the main script run.
+Therefore, we need to define the path to it.
+
+Second, we need to establish the NeMo Curator script we want to run.
+This can be a command line utility like ``text_cleaning`` we have above, or it can be your own custom script ran with ``python path/to/script.py``
+
+
+Finally, we combine all of these into a ``SlurmJobConfig``. This config has many options for configuring the Dask cluster.
+We'll highlight a couple of important ones:
+
+* ``device="cpu"`` determines the type of Dask cluster to initialize. If you are using GPU modules, please set this equal to ``"gpu"``.
+* ``interface="etho0"`` specifies the network interface to use for communication within the Dask cluster. It will likely be different for your Slurm cluster, so please modify as needed.
```

Comment:
It might be helpful to have more about commands to use to figure out what to set for this?